Better control over Pagination stops

IJMacD opened the pull request:
I had the case where I needed to hide the last pages of pagination (the very oldest posts).

I found the `end_step` (not documented anywhere) controlled how many pages are shown at either end of the pagination range.
Additionally I was not able to set this to zero to hide the pages completely.

My proposal is to allow zero pages to be shown at the ends of the list. Also to enable different values for the start and end. The start value defaults to the same as the end value for backwards compatibility.

<!--
First off, hello!

Thanks for submitting a PR. We love/welcome PRs (especially if it's your first).
Have any questions? Read this section in CONTRIBUTING.md: https://github.com/timber/timber/blob/master/CONTRIBUTING.md#pull-requests.
--> 

**Ticket**: # <!-- Ignore this if not relevant -->

## Issue
<!-- Description of the problem that this code change is solving -->
 I needed to hide the last pages of pagination (the very oldest posts).

## Solution
<!-- Description of the solution that this code changes are introducing to the application. -->
My solution is to add a new optional parameter to the paginate_links arg array allowing control over the start and end separately.
Also allow zero as a valid value for the ends of the list.

## Impact
<!-- What impact will this have on the current codebase, performance, backwards compatibility? -->
The new parameter defaults to the value of the old one so maintains perfect backwards compatibility.

## Usage Changes
<!-- Are there are any usage changes that we need to know about? If so, list them here so that we can integrate it in the release notes and developers know what usage changes are associated to your PR.

Alternatively, you’re very welcome to directly edit the readme.txt file with:
- A quick summary, including your Github handle.
- A list of changes for Theme Developers (under the "Changes for Theme Developers" label).
- New usage instructions, possibly with a short code example.
-->


## Considerations
<!-- As we do not live in an ideal world it's worth to share your thought on how we could make the solution even better. -->


## Testing
<!-- Are unit tests included? If they need to be written, please provide pseudo code for a scenario that fails without your code, but succeeds with it -->

jarednova commented:
@IJMacD thanks for the PR! Would you be up for writing some tests in `TestTimberPagination` to cover the new parameters you added?

The good news is that it seems that nothing about the prior behaviors are broken. While Travis is failing right now, it's due to an unrelated issue about changes in Twig (#2301)

IJMacD commented:
I've added the tests and checked against the original code. I found that setting `mid_size` to 0 was previously permitted yet would cause a PHP Notice about an 'Undefined index: link' causing the test suite to fail so I fixed that too.

I was aware of #2301. If you need to see green ticks from the tests I can rebase if you'd like?

jarednova commented:
This is great, thanks @IJMacD ! Really appreciate all the permutations of tests/sizes so we can make sure things (keep) behaving into the future.
